commons/commons-serialization/src/main/kotlin/ac/uk/ebi/biostd/integration/SubFormat.kt

```diff
@@ -1,5 +1,42 @@
 package ac.uk.ebi.biostd.integration
 
-enum class SubFormat {
-    XML, JSON, TSV, JSON_PRETTY
+import ac.uk.ebi.biostd.exception.InvalidExtensionException
+
+sealed class SubFormat {
+
+    companion object {
+        fun valueOf(format: String): SubFormat {
+            return when (format) {
+                "TSV" -> PlainTsv
+                "XML" -> XmlFormat
+                "JSON" -> PlainJson
+                else -> throw InvalidExtensionException(format)
+            }
+        }
+
+        fun fromExtension(extension: String): SubFormat {
+            return when (extension) {
+                "tsv" -> PlainTsv
+                "xlsx" -> XlsxTsv
+                "xml" -> XmlFormat
+                "json" -> PlainJson
+                else -> throw InvalidExtensionException(extension)
+            }
+        }
+
+        val TSV = PlainTsv
```

Comment:
I think it should be just `Tsv` since "Plain" is the only type, I mean, all TSV is plain TSV